org.eclipse.jdt.launching/launching/org/eclipse/jdt/internal/launching/environments/DefaultAccessRuleParticipant.java

```diff
@@ -54,19 +51,19 @@ public IAccessRule[][] getAccessRules(IExecutionEnvironment environment, IVMInst
 				return new IAccessRule[0][]; // in 9+ access rules are superseded by limit-modules
 			}
 		}
-		IAccessRule[][] allRules = null;
-		allRules = fgRules.get(environment.getId());
+		IAccessRule[][] allRules = fgRules.get(environment.getId());
 		if (allRules == null || allRules.length != libraries.length) {
 			// if a different number of libraries, create a new set of rules
 			String[] packages = retrieveSystemPackages(environment);
-			IAccessRule[] packageRules = null;
+			IAccessRule[] packageRules;
 			if (packages.length > 0) {
 				packageRules = new IAccessRule[packages.length + 1];
 				for (int i = 0; i < packages.length; i++) {
-					packageRules[i] = JavaCore.newAccessRule(new Path(packages[i].replace('.', IPath.SEPARATOR)), IAccessRule.K_ACCESSIBLE);
+					IPath pattern = IPath.fromOSString(packages[i].replace('.', IPath.SEPARATOR));
```

Comment:
`fromOSString` -> `fromPortableString` ?

Reply:
It used to be `new Path` and `fromOSString()` just calls that. But eventually the only difference will probably be that on Windows the IS_WINDOWS flag within the Path implementation is set.
But I guess that does not have any real effect.

-> Applied your suggestion.